package fetch.cfg.device;

message NIDAQShutter
{ 
  optional uint32 open       = 1 [default=1];                   // set point for an open shutter
  optional uint32 closed     = 2 [default=0];                   // set point for an open shutter 
  optional string do_channel = 3 [default="/Dev1/port0/line8"]; // control channel
}

message SimulatedShutter
{ optional bool state = 1 [default=false];
}

message vDAQShutter
{ 
  optional uint32 open       = 1 [default=1];                   // set point for an open shutter
  optional uint32 closed     = 2 [default=0];                   // set point for an open shutter 
  optional int32  device_num = 3 [default=0];                   // vDAQ device number
  optional string do_channel = 4 [default="D2.4"];              // control channel
}

message Shutter
{
  enum ShutterType
  {
    Simulated = 0;
    NIDAQ     = 1;
    vDAQ      = 2;
  }
  optional ShutterType  kind  = 1 [default=vDAQ];
  optional NIDAQShutter nidaq = 2;
  optional SimulatedShutter simulated = 3;
  optional vDAQShutter vdaq = 4;
}